docs/ORT_use_trtion_kernel.md

```diff
@@ -0,0 +1,84 @@
+## Description
+In some scenarios, the triton written kernels are more performant than CK or other handwritten kernels, so we implement a framework that onnxruntime can use these triton written kernels.
+
+Here we use `softmax` op as an example to show how to integrate a triton written kernel into onnxruntime cuda/rocm EP.
+
+#### Write and compile triton kernel
+We have implemented a softmax kernel using triton at `onnxruntime/core/providers/rocm/math/softmax.py`
+
+```
+@triton.jit
```

Comment:
Can we add a new `onnxruntime.trition.compiled_metadata` decorator that maintain the metadata, add some attribute to the function, say `softmax_kernel.onnxruntime_compiled_metadata` is accessible. Then the compiler driver `compile_triton.py` queries all functions that are annotated, use the metadata to drive the compilation. This gives us a lot of flexibility.

Reply:
```py
@onnxruntime.trition.compiled_metadata(
  dtypes = ['fp32', 'fp16'],
  blocks = [1024, 2048, 4096, 8192, 16384],
  name_pattern = 'softmax_{}_{}',
  sig_pattern = '*{},*{},i32,i32,i32',
  group_pattern = 'softmax_{}',
)
@triton.jit
def softmax_kernel(...): ...
```

Reply:
Now Triton is used in ORT in two ways,
1. as a compiler, where you can compile a ort kernel and its subgraph to a optimized kernel for GPUs
2. as a performant kernel development tool, where you can use it to develop performant kernels like nvcc.

I guess your reference was to use triton as a backend compiler.
This PR is for the 2nd usage, where it is used a kernel development tool by leveraging the kernel explorer's capabilities.